src/rc/seedrng.c

```diff
@@ -404,17 +473,19 @@ int main(int argc _unused, char *argv[] _unused)
 	blake2s_update(&hash, &realtime, sizeof(realtime));
 	blake2s_update(&hash, &boottime, sizeof(boottime));
 
-	if (mkdir(SEED_DIR, 0700) < 0 && errno != EEXIST)
-		eerrorx("Unable to create \"%s\" directory: %s", SEED_DIR, strerror(errno));
+	if (mkdir(seed_dir, 0700) < 0 && errno != EEXIST)
+		eerrorx("Unable to create \"%s\" directory: %s", seed_dir, strerror(errno));
 
-	lock = open(LOCK_FILE, O_WRONLY | O_CREAT, 0000);
+	lock = open(lock_file, O_WRONLY | O_CREAT, 0000);
 	if (lock < 0 || flock(lock, LOCK_EX) < 0)
 		eerrorx("Unable to open lock file: %s", strerror(errno));
```

Comment:
https://github.com/OpenRC/openrc/pull/509 handles the case of closing lock where lock >= 0, but flock() < 0, if you care about closing everything before the OS does it for you.

Reply:
"It looks like" is what I said.

Anyway, please do read #509; it does this more correctly (e.g. handles the case where somebody closes all fds and then exec()s), and fixes the flock thing too.

(Note that none of this actually matters, because it's a short lived utility where the OS cleans up all the resources anyway. But I can understand the preference for cleanliness in exit paths.)